pkg/envtest/apiserver.go

```diff
@@ -416,19 +417,26 @@ func apiServiceNameForSchemeGroupVersion(gv schema.GroupVersion) string {
 
 // Stop stops this GardenerAPIServer and cleans its temporary resources.
 func (g *GardenerAPIServer) Stop() error {
-	// trigger stop procedure
-	g.terminateFunc()
+	var errList []error
 
-	select {
-	case <-g.exited:
-		break
-	case <-time.After(g.StopTimeout):
-		return fmt.Errorf("timeout waiting for gardener-apiserver to stop")
+	// trigger stop procedure
+	if g.terminateFunc == nil {
```

Comment:
Q: shouldn't this line be `g.terminateFunc != nil`?

I guess when it is written like `g.terminateFunc == nil`, it will always lead to segfault when g.terminateFunc is nil.

```suggestion
	if g.terminateFunc != nil {
```

Reply:
Most definitely, thanks for catching!
Added to the respective commit.